src/main/java/com/ltonetwork/seasalt/sign/ECDSA.java

```diff
@@ -0,0 +1,127 @@
+package com.ltonetwork.seasalt.sign;
+
+import com.ltonetwork.seasalt.KeyPair;
+import org.bouncycastle.asn1.ASN1InputStream;
+import org.bouncycastle.asn1.ASN1Integer;
+import org.bouncycastle.asn1.DERSequenceGenerator;
+import org.bouncycastle.asn1.DLSequence;
+import org.bouncycastle.asn1.x9.X9ECParameters;
+import org.bouncycastle.crypto.AsymmetricCipherKeyPair;
+import org.bouncycastle.crypto.Digest;
+import org.bouncycastle.crypto.digests.SHA256Digest;
+import org.bouncycastle.crypto.generators.ECKeyPairGenerator;
+import org.bouncycastle.crypto.params.ECDomainParameters;
+import org.bouncycastle.crypto.params.ECKeyGenerationParameters;
+import org.bouncycastle.crypto.params.ECPrivateKeyParameters;
+import org.bouncycastle.crypto.params.ECPublicKeyParameters;
+import org.bouncycastle.crypto.signers.ECDSASigner;
+import org.bouncycastle.crypto.signers.HMacDSAKCalculator;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.SecureRandom;
+
+public class ECDSA implements Signer {
+    final X9ECParameters curve;
+    final ECDomainParameters domain;
+    final BigInteger HALF_CURVE_ORDER;
+
+    public ECDSA(X9ECParameters curve) {
+        this.curve = curve;
+        this.domain = new ECDomainParameters(curve.getCurve(), curve.getG(), curve.getN(), curve.getH());
+        this.HALF_CURVE_ORDER = curve.getN().shiftRight(1);
+    }
+
+    public KeyPair keyPair() {
+        SecureRandom srSeed = new SecureRandom();
+        byte[] privateKey = generatePrivateKey(srSeed);
+        byte[] publicKey = privateToPublic(privateKey);
+        return new KeyPair(publicKey, privateKey);
+    }
+
+    public KeyPair keyPairFromSeed(byte[] seed) {
+        SecureRandom srSeed = new SecureRandom(seed);
+        byte[] privateKey = generatePrivateKey(srSeed);
+        byte[] publicKey = privateToPublic(privateKey);
+        return new KeyPair(publicKey, privateKey);
+    }
+
+    public KeyPair keyPairFromSecretKey(byte[] privateKey) {
+        byte[] publicKey = privateToPublic(privateKey);
+        return new KeyPair(publicKey, privateKey);
+    }
+
+    public byte[] signDetached(byte[] msg, byte[] privateKey, Digest digest) {
```

Comment:
Digest should be passed to the constructor instead of the sign methods. The object should only have the methods as defined in the `Signer` interface.

Overload the constructor, so it uses sha256 as default.